elastic/apm-server
# testing: Enable stack monitoring local testing

## Motivation/summary

This PR adds a few things to make it easier to test things under the
Elastic Agent in a semi-automated manner, using `systemtest/cmd/runapm`
to create a new Elastic Agent policy with a locally built `apm-server`
binary.
Also adds the ability to enable stack monitoring locally via a new
`metricbeat` container that is only added when `docker-compose --profile
monitoring` is set, with two scripts to make this possible:

- `testing/docker/fleet-server/agent-monitoring.sh` is run inside the
container and modifies the agent configuration to enable monitoring and
publish the metrics in all interfaces.
- `testing/stack-monitoring.sh` can be run from your machine to start
the stack with Stack Monitoring enabled.

`TESTING.md` has been update with all the necessary instructions.

## Checklist



~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)~
~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)`~

## How to test these changes

1. See the `TESTING.md` document.
2. Run `./testing/stack-monitoring.sh` and verify that Elasticsearch and APM Server show metrics in the Stack Monitoring UI.

## Related issues

None, but we want to improve our ability to test things locally.
